### PR TITLE
AzureRm UI part-1

### PR DESCRIPTION
**Pull request by vijay8451**
- Added tests around Finish template and userdata(could-init) 
- Nailgun/Airgun support PRs:
https://github.com/SatelliteQE/nailgun/pull/711
https://github.com/SatelliteQE/nailgun/pull/712
https://github.com/SatelliteQE/airgun/pull/470

- Test results:
```                                                                                                                                                       
tests/foreman/ui/test_computeresource_azurerm.py::test_positive_end_to_end_azurerm_host_provisioned_ft PASSED                                                         [100%]
tests/foreman/ui/test_computeresource_azurerm.py::test_positive_azurerm_host_provision_ud PASSED                                                                      [100%]

```

### CODE REVIEW

File: tests/foreman/ui/test_computeresource_azurerm.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+    :CaseLevel: System
+    """
+
+    hostname = gen_string('alpha')
+    fqdn = '{}.{}'.format(hostname, module_domain.name).lower()
+
+    with session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
```
Comment (mirekdlugosz) [inline tests/foreman/ui/test_computeresource_azurerm.py]: Are these two lines required?

`session` fixture should return you an user that is automatically assigned to organization and location created by these fixtures.

File: tests/foreman/ui/test_computeresource_azurerm.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
```
Comment (mirekdlugosz) [inline tests/foreman/ui/test_computeresource_azurerm.py]: Same as above

File: tests/foreman/ui/test_computeresource_azurerm.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+                {
+                    'host.name': hostname,
+                    'host.hostgroup': module_azure_hg.name,
+                    'provider_content.operating_system.root_password': gen_string('alpha'),
+                    'provider_content.operating_system.image': module_azurerm_finishimg.name,
+                }
+            )
+
+            host_info = session.host.get_details(fqdn)
+            assert session.host.search(fqdn)[0]['Name'] == fqdn
```
Comment (mirekdlugosz) [inline tests/foreman/ui/test_computeresource_azurerm.py]: `host.get_details()` uses `.search()` during `step`, so this assertion is redundant - it must pass if `get_details()` succeeded.

File: tests/foreman/ui/test_computeresource_azurerm.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+            assert azurecloud_vm.ip == host_info['properties']['properties_table']['IP Address']
+            assert azurecloud_vm.type == AZURERM_VM_SIZE_DEFAULT
+
+            # Host Delete
+            message = session.host.delete(fqdn)
+            assert (
+                'Are you sure you want to delete host {}? This will delete the VM and '
+                'its disks, and is irreversible. This behavior can be changed via '
+                'global setting "Destroy associated VM on host delete".'.format(fqdn)
+            ) == message
```
Comment (mirekdlugosz) [inline tests/foreman/ui/test_computeresource_azurerm.py]: Do we want this assertion here?
Seems we have that already covered in other tests. And I wouldn't fail this test if exact wording ever changes - seems a little too aggressive for my taste.

File: tests/foreman/ui/test_computeresource_azurerm.py [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                {
+                    'host.name': hostname,
+                    'host.hostgroup': module_azure_hg.name,
+                    'provider_content.operating_system.root_password': gen_string('alpha'),
+                    'provider_content.operating_system.image': module_azurerm_cloudimg.name,
+                }
+            )
+
+            host_info = session.host.get_details(fqdn)
+            assert session.host.search(fqdn)[0]['Name'] == fqdn
```
Comment (mirekdlugosz) [inline tests/foreman/ui/test_computeresource_azurerm.py]: Same comment about redundancy here.

File: tests/foreman/ui/test_computeresource_azurerm.py (hunk 1 above)
Comment (vijay8451) [reply tests/foreman/ui/test_computeresource_azurerm.py]: yeah , we do not required that ...removed in latest push.

File: tests/foreman/ui/test_computeresource_azurerm.py (hunk 3 above)
Comment (vijay8451) [reply tests/foreman/ui/test_computeresource_azurerm.py]: good catch , removed that assertion as not required and make sense much here.

File: tests/foreman/ui/test_computeresource_azurerm.py (hunk 4 above)
Comment (vijay8451) [reply tests/foreman/ui/test_computeresource_azurerm.py]: removed

**Comment by vijay8451** [comment]
> Left some improvement suggestions.

@mirekdlugosz revisit pls now.

File: tests/foreman/ui/test_computeresource_azurerm.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+            assert azurecloud_vm.name == hostname.lower()
+            assert azurecloud_vm.ip == host_info['properties']['properties_table']['IP Address']
+            assert azurecloud_vm.type == AZURERM_VM_SIZE_DEFAULT
+
+        except Exception as error:
+            azure_vm = entities.Host().search(query={'search': 'name={}'.format(fqdn)})
+            if azure_vm:
+                azure_vm[0].delete()
+            raise error
+
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_computeresource_azurerm.py]: I'm trying to understand here that if you wanted to raise anyway, you could have only had try and finally, by that way you no need to raise the error and checking again and again about the  `azure_vm = entities.Host().search(query={'search': 'name={}'.format(fqdn)})    `

I might be wrong, I'm not able to understand that except block need.

Comment (vijay8451) [reply tests/foreman/ui/test_computeresource_azurerm.py]: Probably a copy/paste error but not sure why i kept it under same ... I have removed that now as not require :)

**Pull request merged**